api/bootstrap/kubeadm/v1beta2/kubeadm_types.go

```diff
@@ -207,6 +207,11 @@ type ClusterConfiguration struct {
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=63
 	ClusterName string `json:"clusterName,omitempty"`
+
+	// CertificateValidityPeriod specifies the validity period for a non-CA certificate generated by kubeadm.
+	// Default value: 8760h (365 days * 24 hours = 1 year)
+	// +optional
+	CertificateValidityPeriod *metav1.Duration `json:"certificateValidityPeriod,omitempty"`
```

Comment:
I see a PR to convert Durations to seconds https://github.com/kubernetes-sigs/cluster-api/pull/12327, should this be handled in the same or should we keep it in Duration format to keep it like kubeadm [way](https://github.com/kubernetes/kubernetes/blob/11c5a4517122a45a22174e962db23fbdfc5ad67d/cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go#L159)

Reply:
Sure, thanks